sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -37,6 +37,20 @@ namespace Azure { namespace Core { namespace Http {
    *@brief Used to read data to/from a service.
    */
   class BodyStream {
+  protected:
+    /**
+     * @brief Read portion of data into a buffer.
+     *
+     * @remark This is the Read implementation that all derived classes need to provide.
+     *
+     * @param context #Context so that operation can be canceled.
+     * @param buffer Pointer to a byte buffer to read the data into.
+     * @param count Size of the buffer to read the data into.
+     *
+     * @return Number of bytes read.
+     */
+    virtual int64_t OnRead(Context const& context, uint8_t* buffer, int64_t count) = 0;
```

Comment:
Make this `private`, or is there a reason to keep it `protected`?
```suggestion
  private:
    /**
     * @brief Read portion of data into a buffer.
     *
     * @remark This is the Read implementation that all derived classes need to provide.
     *
     * @param context #Context so that operation can be canceled.
     * @param buffer Pointer to a byte buffer to read the data into.
     * @param count Size of the buffer to read the data into.
     *
     * @return Number of bytes read.
     */
    virtual int64_t OnRead(Context const& context, uint8_t* buffer, int64_t count) = 0;
```